addons/mrp/models/mrp_routing.py

```diff
@@ -49,14 +50,24 @@ def _compute_time_cycle(self):
         for operation in manual_ops:
             operation.time_cycle = operation.time_cycle_manual
         for operation in self - manual_ops:
-            data = self.env['mrp.workorder'].read_group([
+            data = self.env['mrp.workorder'].search([
                 ('operation_id', '=', operation.id),
                 ('qty_produced', '>', 0),
-                ('state', '=', 'done')], ['operation_id', 'duration', 'qty_produced'], ['operation_id'],
-                limit=operation.time_mode_batch)
-            count_data = dict((item['operation_id'][0], (item['duration'], item['qty_produced'])) for item in data)
-            if count_data.get(operation.id) and count_data[operation.id][1]:
-                operation.time_cycle = (count_data[operation.id][0] / count_data[operation.id][1]) * (operation.workcenter_id.capacity or 1.0)
+                ('state', '=', 'done')],
+                limit=operation.time_mode_batch,
+                order="date_finished desc")
+            # To compute the time_cycle, we can take the total duration of previous operations
+            # but for the quantity, we will take in consideration the qty_produced like if the capacity was 1.
+            # So producing 50 in 00:10 with capacity 2, for the time_cycle, we assume it is 25 in 00:10
+            # When recomputing the expected duration, the capacity is used again to divide the qty to produce
+            # so that if we need 50 with capacity 2, it will compute the expected of 25 which is 00:10
+            total_duration = 0  # Can be 0 since it's not an invalid duration for BoM
+            total_qty_per_capacity_for_duration = 0  # Never 0 unless infinite item['workcenter_id'].capacity
+            for item in data:
+                total_duration += item['duration']
+                total_qty_per_capacity_for_duration += math.ceil(item['qty_produced'] / item['workcenter_id'].capacity or 1.0)
```

Comment:
You could use float_round with rounding_method='DOWN' to use odoo tools and avoid a bugifx later on a strange rounding issue

Reply:
@amoyaux Don't you mean rounding_method='UP' ? Since we are using math.ceil (1.001 becomes 2)
It is also used here https://github.com/odoo-dev/odoo/blob/11c02365d52c973b6c58685cbe52cf3ea94c5fd0/addons/mrp/models/mrp_workorder.py#L691 so I changed my variable names too to line up with the code to see more easily the link between the two (divide by average cycle_number in mrp_routing.py and multiply by workorder cycle_number in mrp_workorder.py